src/seedu/addressbook/data/person/Address.java

```diff
@@ -25,7 +29,25 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+        
+        int i = 0;
+        for(String subAddress : address.split(",")){
+        	if(i == 0){
+        		block = new Block(subAddress);
+        	}
+        	else if(i == 1){
+        		street = new Street(subAddress);
+        	}
+        	else if(i == 2){
+        		unit = new Unit(subAddress);
+        	}
+        	else if(i == 3){
+        		portalCode = new PortalCode(subAddress);
+        	}
+            i++;
+        }
```

Comment:
Check out our coding convention for writing conditional statement. BTW, `subAddress` doesn't give any meaning, maybe that's also because you use magic number instead of constant variable.